pkg/config/config.go

```diff
@@ -64,19 +64,32 @@ type Config struct {
 
 // TODO: https://github.com/jbrekelmans/kube-compose/issues/64
 // nolint
-func New() (*Config, error) {
-	fileName := "docker-compose.yml"
-	data, err := ioutil.ReadFile(fileName)
-	if err != nil {
+func New(file *string) (*Config, error) {
+	var data []byte
+	var err error
+	var fileName string
+	if file != nil {
+		data, err = ioutil.ReadFile(*file)
 		if os.IsNotExist(err) {
-			fileName = "docker-compose.yaml"
-			data, err = ioutil.ReadFile(fileName)
+			return nil, errors.Wrap(err, fmt.Sprintf("error loading docker-compose file at %#v", fileName))
 		}
 		if err != nil {
 			return nil, err
 		}
+		fileName = *file
+	} else {
+		fileName = "docker-compose.yml"
+		data, err = ioutil.ReadFile(fileName)
+		if err != nil {
+			if os.IsNotExist(err) {
+				fileName = "docker-compose.yaml"
+				data, err = ioutil.ReadFile(fileName)
+			}
+			if err != nil {
+				return nil, errors.Wrap(err, fmt.Sprintf("error loading %#v in the current working directory", fileName))
```

Comment:
I don't like the *in the current working directory* part of the error, IMO the error should be the same as above. In that case we can actually move the `err != nil` check and further deduplicate.